bam_sort.c

```diff
@@ -1853,15 +1854,24 @@ static int ks_radixsort(size_t n, bam1_tag *buf, const sam_hdr_t *h)
     for (; max_tid > 0; max_tid >>= 8) tid_bytes++;
     assert(pos_bytes + tid_bytes < sizeof(buf[0].u.pos_tid));
 
+    tid_shift_l = pos_bytes * 8;
+    tid_shift_r = 64 - pos_bytes;
```

Comment:
Shouldn't this be `64 - tid_shift_l`?

Reply:
Good spot.  Should now be fixed with an enhanced test to catch the bug.